analysis/datetime/flexible/flexible.go

```diff
@@ -44,6 +45,52 @@ func (p *DateTimeParser) ParseDateTime(input string) (time.Time, error) {
 	return time.Time{}, analysis.ErrInvalidDateTime
 }
 
+func validateLayout(layout string) bool {
+	validMagicNumbers := map[string]bool{
+		"2006":    true,
+		"06":      true, // Year
+		"01":      true,
+		"1":       true,
+		"_1":      true,
+		"January": true,
+		"Jan":     true, // Month
+		"02":      true,
+		"2":       true,
+		"_2":      true,
+		"Monday":  true,
+		"Mon":     true, // Day
+		"15":      true,
+		"3":       true,
+		"03":      true, // Hour
+		"4":       true,
+		"04":      true, // Minute
+		"5":       true,
+		"05":      true, // Second
+		"PM":      true,
+		"pm":      true,
+		"MST":     true,
+		"Z0700":   true, // prints Z for UTC
+		"Z070000": true,
+		"Z07":     true,
+		"0700":    true,
+		"070000":  true,
+		"07":      true,
+		"":        true,
+	}
+	re := regexp.MustCompile("[- :T,\\.<>;\\?!`~@#$%\\^&\\*|\\(\\){}\\[\\]/\\\\]")
```

Comment:
Same for the regexp, move the initialization and all the common stuff to outside this method.